libs/librepcb/projecteditor/boardeditor/fsm/bes_select.cpp

```diff
@@ -759,6 +771,91 @@ bool BES_Select::removeSelectedItems() noexcept {
   }
 }
 
+void BES_Select::measureLengthInDirection(bool              directionBackwards,
+                                          const BI_NetLine& netline,
+                                          QSet<Uuid>&       visitedSegments,
+                                          UnsignedLength&   totalLength) {
+  auto currentNetline = std::cref(netline);
+  auto currentAnchor  = std::cref(directionBackwards ? netline.getStartPoint()
+                                                    : netline.getEndPoint());
+
+  for (;;) {
+    tl::optional<BI_NetLine&> nextNetline = tl::nullopt;
+    for (BI_NetLine* nl : currentAnchor.get().getNetLines()) {
```

Comment:
I guess here you could use `const`(?):

```suggestion
    for (const BI_NetLine* nl : currentAnchor.get().getNetLines()) {
```